Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -28,20 +29,107 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
-        {
+        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
+        {
             var result = SelectMeals(protein, carbs, fat, dietPlans).SequenceEqual(expected) ? "PASS" : "FAIL";
             Console.WriteLine($"Proteins = [{string.Join(", ", protein)}]");
             Console.WriteLine($"Carbs = [{string.Join(", ", carbs)}]");
-            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
+            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
             Console.WriteLine($"Diet plan = [{string.Join(", ", dietPlans)}]");
-            Console.WriteLine(result);
+            Console.WriteLine(result);
         }
+        public static List<int> Maxm(int[] arr, List<int> count)
+        {
+            int maxm = arr[count[0]];
+            List<int> temp = new List<int>();
+
+            for(int i=0; i< count.Count; i++)
+            {
+                if (arr[count[i]] > maxm)
+                    maxm = arr[count[i]];
+            }
+            for(int i = 0; i < count.Count; i++)
+            {
+                if (arr[count[i]] == maxm)
+                    temp.Add(count[i]);
+            }
+            
+            return temp;
+        }
+        public static List<int> Minm(int[] arr, List<int> count)
+        {
+            int minm = arr[count[0]];
+            List<int> temp = new List<int>();
 
-        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
-        {
+            for (int i = 0; i < count.Count; i++)
+            {
+                if (arr[count[i]] < minm)
+                    minm = arr[count[i]];
+            }
+            for (int i = 0; i < count.Count; i++)
+            {
+                if (arr[count[i]] == minm)
+                    temp.Add(count[i]);
+            }
+            
+            return temp;
+        }
+        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            List<int> count = new List<int>();
+            int[] ans = new int[dietPlans.Length];
+            int[] total = new int[protein.Length];
+
+            for(int i=0; i<protein.Length; i++)

```

Comment:
Instead of using same for loop in multiple places, try to add logic in a single one.
This will reduce code size and execution time